src/Sulu/Bundle/AdminBundle/Admin/Routing/FormRouteBuilderTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Admin\Routing;
+
+trait FormRouteBuilderTrait
```

Comment:
should we move this methods to the `RouteBuilderTrait` and just use that one everywhere?
right now it is not really intuitive where a new method should be added if it is used in multiple `RouteBuilder` classes. (eg. `setEditRoute`)
also it is not intuitive if a method should be added if it is used only in a single `RouteBuilder` class. (eg. `setPreviewCondition`)

Reply:
when using a single `RouteBuilderTrait` (or maybe `RouteOptionsTrait`), that trait would be the single source of truth for all option keys. this would force us to keep the option-keys consistent between different views.

Reply:
Doesn't that feel to you it would couple the code more than necessary? My main reason behind this idea was to avoid breaking the `PreviewForm` route definitions when something in the normal `Form` changes (and the same for the `FormOverlayList` route). I think I would still maybe extract a `TabRouteBuilderTrait` for the tab specific stuff (since this is really also the same for every view), and remove the `RouteBuilderTrait` completely, and move the functions from it either to the `ListRouteBuilderTrait` or the `FormRouteBuilderTrait` - or to both of them. I think duplicating the logic of e.g. `setResourceKey` does not hurt, since it is pretty simple. In addition to that this solution would be sufficient to guarantee that the `PreviewForm` and `FormOverlayList` will continue to work if the views they are using changed some of its route options.

Reply:
alright - removed the general `RouteBuilderTrait` and added a `TabRouteBuilderTrait` 🙂